File: 682-Baseball-Game.cpp
class Solution {
public:
    int calPoints(vector<string>& operations) {
        stack<int> records;
        int sum = 0;

        for (int i = 0; i < operations.size(); i++) {
            if (operations[i] == \C\) {
                records.pop();
            } else if (operations[i] == \D\) {
                records.push(records.top() * 2);
            } else if (operations[i] == \+\) {
                int top = records.top();
                records.pop();
                int newTop = records.top() + top;
                records.push(top);
                records.push(newTop);
            } else {
                records.push(stoi(operations[i]));
            }
        }

        while (!records.empty()) {
            sum += records.top();
            records.pop();
        }
        return sum;
    }
};

File: 682-Baseball-Game.js
/**
 * @param {string[]} operations
 * @return {number}
 */
var calPoints = function (operations) {
    let record = [];

    for (let i = 0; i < operations.length; i++) {
        if (Number(operations[i]))
            record.push(Number(operations[i]));
        else if (operations[i] === 'C')
            record.pop();
        else if (operations[i] === 'D')
            record.push(record[record.length - 1] * 2);
        else if (operations[i] === '+')
            record.push(record[record.length - 1] + record[record.length - 2]);
    }

    return record.reduce((acc, curr) => acc + curr, 0);
};